### PR TITLE
Fix bad smell

### PR DESCRIPTION
**Pull request by koolay**

### CODE REVIEW

**Comment by weiguoz** [comment]
Thank you @koolay . Would you please sign the [CLA](https://www.clahub.com/agreements/sql-machine-learning/gomaxcompute) if you want to check in the code?
FYI: [What's CLA](https://zhuanlan.zhihu.com/p/68251730)

**Comment by koolay** [comment]
> Thank you @koolay . Would you please sign the [CLA](https://www.clahub.com/agreements/sql-machine-learning/gomaxcompute) if you want to check in the code?
> FYI: [What's CLA](https://zhuanlan.zhihu.com/p/68251730)

Done.

File: .golangci.yml [hunk 1]
```diff
@@ -0,0 +1,200 @@
+run:
```
Comment (tonyyang-svail) [inline .golangci.yml:1]: What is the usage of this file?

File: client.go [hunk 2]
```diff
@@ -132,7 +130,7 @@ func (conn *odpsConn) sign(r *http.Request) {
 			}
 			canonicalResource.WriteString(k)
 			v := urlParams[k]
-			if v != nil && len(v) > 0 && v[0] != "" {
```
Comment (tonyyang-svail) [inline client.go:135]: Why removing the `v != nil` check?

File: dsn.go [hunk 3]
```diff
@@ -9,8 +9,8 @@ import (
 
 var (
 	// Regexp syntax: https://github.com/google/re2/wiki/Syntax
-	reDSN   = regexp.MustCompile(`^([a-zA-Z0-9_-]+):([=a-zA-Z0-9_-]+)@([:a-zA-Z0-9/_.-]+)\?([^/]+)$`)
-	reQuery = regexp.MustCompile(`^([a-zA-Z0-9_-]+)=([a-zA-Z0-9_-]*)$`)
+	// reQuery = regexp.MustCompile(`^([a-zA-Z0-9_-]+)=([a-zA-Z0-9_-]*)$`)
```
Comment (tonyyang-svail) [inline dsn.go]: Maybe remove commented code.

**Review by tonyyang-svail — COMMENTED** [review]
Thank @koolay for submitting this PR. I left a few comments.

File: .golangci.yml (hunk 1 above)
Comment (weiguoz) [inline .golangci.yml:1]: Would you please add the reference? It looks like this file cited externally.

Comment (koolay) [reply .golangci.yml:1]: `.golangci.yml `  required by https://github.com/golangci/golangci-lint

Comment (koolay) [reply .golangci.yml:1]: it's required for local linting.  
[golangci-lint](https://github.com/golangci/golangci-lint)

File: client.go (hunk 2 above)
Comment (koolay) [reply client.go:135]: `if  len(v) > 0` works same as `if v != nil && len(v) > 0`

**Comment by tonyyang-svail** [comment]
Hi @weiguoz, to test the execution, could you please file an identical branch to our origin repo?

**Comment by weiguoz** [comment]
Thanks to @koolay.
I move forward this PR because of the CI(#56 ) passed.

**Pull request merged**